pkg/controller/health/reconciler.go

```diff
@@ -127,6 +120,11 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			unstructuredObj.SetAPIVersion(ref.APIVersion)
 			unstructuredObj.SetKind(ref.Kind)
 			obj = unstructuredObj
+		} else {
+			var ok bool
+			if obj, ok = runtimeObject.(client.Object); !ok {
+				return ctrl.Result{}, fmt.Errorf("could not execute health check because object type %q is unsupported", ref.GroupVersionKind().String())
```

Comment:
Hmm, this will cause the MR to be requeued with exponential backoff for the health check.
I think, we generally would want to avoid that, right?

Reply:
Makes sense, thank you! I've changed it.